client/app/lib/kivy_manager.py

```diff
@@ -1,23 +1,80 @@
 import asyncio
 from typing import Any
 
+import kivymd.uix.button
 import websockets
 from kivy import Logger
-from kivy.app import App
+from kivy.core.window import Window
+from kivy.lang.builder import Builder
+from kivy.modules import inspector
+from kivy.utils import get_color_from_hex
+from kivymd.app import MDApp
+from kivymd.uix.floatlayout import MDFloatLayout
 
+from ..utils import Colors, app_dir
 
-class ClientUI(App):
+Window.borderless = True
+Window.custom_titlebar = True
+
+
+class TitleBar(MDFloatLayout):
+    """Custom TitleBar for the app"""
+
+    md_bg_color = get_color_from_hex(Colors.accent_bg.value)
+    button_bg = get_color_from_hex(Colors.primary_bg.value)
+    button_size = "15sp"
+
+    def __init__(self):
+        super(TitleBar, self).__init__()
+        self.app: MDApp = MDApp.get_running_app()
+
+    def handle_buttons(self, instance: kivymd.uix.button.BaseButton):
+        """Callback Function for all buttons in titlebar
+
+        :param instance a button object that is a subclass of kivymd.uix.button.BaseButton
+
+        A button object was chosen instead of say a string, so that later on the caller itself can be edited
+        """
+        match_string = None
+        if hasattr(instance, "icon"):
+            match_string = instance.icon
+
+        if match_string:
+            match match_string:
+                case "close":
+                    self.app.stop()
+                case "window-minimize":
+                    self.app.root_window.minimize()
+                case "window-maximize":
+                    self.app.root_window.maximize()
+
+
+class ClientUI(MDApp):
     """Main Class to Build frontend on."""
 
     ws: websockets.WebSocketClientProtocol = None
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
         self.ws_handler_task = None
+        self.title = "Blak"
```

Comment:
build method has init the variable title, unnecessary to init it

Reply:
this is more verbose but ok

Reply:
ok